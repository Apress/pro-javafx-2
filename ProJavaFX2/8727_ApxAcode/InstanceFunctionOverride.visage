/*
 * @test
 * @run
 */

class A {
  function f(i:Integer):Integer {
    println("A.f() is invoked with parameter i = {i}.");
    i + 3
  }
}

class B extends A {
  override function f(i:Integer):Integer {
    println("B.f() is invoked with parameter i = {i}.");
    i * 5;
  }
}

var a = A {};
var b = B {};
var ra = a.f(4);
var rb = b.f(7);
println("a.f(4) = {ra}.");
println("b.f(7) = {rb}.");
