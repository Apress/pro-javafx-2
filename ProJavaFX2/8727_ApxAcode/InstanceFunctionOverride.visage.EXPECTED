A.f() is invoked with parameter i = 4.
B.f() is invoked with parameter i = 7.
a.f(4) = 7.
b.f(7) = 35.
